Fixed grammatical errors in docs (#1106)
Qiskit/qiskit-experiments: docs/tutorials/curve_analysis.rst

3. Fitting
^^^^^^^^^^

> Curve analysis calls :meth:`_run_curve_fit`, method which is the core functionality of the fitting.

Reviewer: ```suggestion
Curve analysis calls the :meth:`_run_curve_fit` method, which is the core functionality of the fitting.
```

Another method :meth:`_generate_fit_guesses` is internally called to
prepare the initial guess and parameter boundary with respect to the formatted data.
Developers usually override this method to provide better initial guesses
tailored to the defined fit model or type of the associated experiment.
See :ref:`curve_analysis_init_guess` for more details.
Developers can also override the entire :meth:`_run_curve_fit` method to apply
custom fitting algorithms. This method must return a :class:`.CurveFitResult` dataclass.

4. Post processing
^^^^^^^^^^^^^^^^^^

Curve analysis runs several postprocessing against the fit outcome.

> It calls :meth:`._create_analysis_results` to create :class:`.AnalysisResultData` class

Reviewer: ```suggestion
It calls :meth:`._create_analysis_results` to create the :class:`.AnalysisResultData` class
```

for the fitting parameters of interest. A developer can inject a custom code to
compute custom quantities based on the raw fit parameters.
See :ref:`curve_analysis_results` for details.

> Afterwards, the analysis draws several curves in the Matplotlib figure.

Author: What's the current way to draw figures?

Users can set a custom plotter in :class:`.CurveAnalysis` classes, to customize
figures, by setting the :attr:`~.CurveAnalysis.plotter` attribute (see documentation of the visualization module).

> Finally, it returns the list of created analysis results and a figure.

Reviewer: How about this?

```
Afterwards, figure plotting is handed over to the :doc:`Visualization </tutorials/visualization>` module via
the :attr:`~.CurveAnalysis.plotter` attribute, and a list of created analysis results and figures are returned.
```

Author: It's very good, but a question: earlier the document says:
```
Finally, the analysis outputs a set of :class:`.AnalysisResultData` entries
for important fit outcomes along with a single figure of the fit curves
with the measured data points.
```
So, do we return `a list of created analysis results and figures` or `a list of created analysis results and a figure`?

Author: Done

.. _curve_analysis_init_guess:

Providing initial guesses
-------------------------

Fitting without initial guesses for parameters often results in a bad fit. Users can
provide initial guesses and boundaries for the fit parameters through analysis options
``p0`` and ``bounds``. These values are the dictionary keyed on the parameter name, and
one can get the list of parameters with the :attr:`CurveAnalysis.parameters`. Each

> boundary value can be a tuple of floats representing min and max value.

Reviewer: ```suggestion
boundary value can be a tuple of floats representing minimum and maximum values.
```
